src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -163,3 +164,18 @@ Blazor.start = boot;
 if (shouldAutoStart()) {
   boot();
 }
+
+function showUnavailableTransportErrorMessage(ex: Error) {
+  const unableToConnectTransportMessage = "Unable to connect to the server with any of the available transports.";
+  if (!ex.message.startsWith(unableToConnectTransportMessage)) {
+    return;
+  }
+
+  if (ex.message.includes(`'WebSockets' is not supported in your environment.`)) {
```

Comment:
Would we be able to change SignalR to throw a type and use that here instead of string sniffing: https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/throw#throw_an_object?

Reply:
cc/ @BrennanConroy would you be open to this change?

Reply:
I'm not against throwing a specific error type at https://github.com/dotnet/aspnetcore/blob/3eb0fa8abe62f9a937a78890762908159174ff91/src/SignalR/clients/ts/signalr/src/HttpConnection.ts#L408

Much better than doing string matching since those can change at any time.

Reply:
Just added a `Error.name` here: 

https://github.com/dotnet/aspnetcore/pull/34644/commits/e03d084445821d142d144a54c535658716fc0f20#diff-757d83bcf707b34888c824c952e4717638803af7b5372f6c5da1eb838f1a5814R450-R452

Due to the way exception [nesting is done in the `HttpConnection`](https://github.com/dotnet/aspnetcore/blob/e03d084445821d142d144a54c535658716fc0f20/src/SignalR/clients/ts/signalr/src/HttpConnection.ts#L408), having the name itself propagate up would likely require a lot more extensive changes. Instead, I'm still relying a _bit_ on string sniffing, but I'm sniffing the custom `Error.name` instead of the message.

https://github.com/dotnet/aspnetcore/pull/34644/commits/e03d084445821d142d144a54c535658716fc0f20#diff-7c47c684c78e3f5b5eeecb28b249b03afcefcd90a925cc20b69ec2cdec261752R134

Reply:
What was wrong with changing the type of exception we throw?

Reply:
If you're referring to here:

https://github.com/dotnet/aspnetcore/blob/3eb0fa8abe62f9a937a78890762908159174ff91/src/SignalR/clients/ts/signalr/src/HttpConnection.ts#L408

Several different issues can result in an exception being thrown there. So changing the exception type being thrown there wouldn't provide us with specific enough information (ie. diagnose a WS failed connection).

Reply:
Updated 895afb22fb.